pkg/network/java/hotspot.go

```diff
@@ -317,5 +352,25 @@ func (h *Hotspot) Attach(agentPath string, args string, uid int, gid int) error
 			loadCommand += " " + args
 		}
 	}
-	return h.command(loadCommand, !isJar)
+
+	hasRetried := false
+retry:
+	if err := h.command(loadCommand, !isJar); err != nil {
+		// if we receive EPIPE from the kernel, let's retry with switching effective uid/gid
+		// see dialunix() for more info
+		if errors.Is(err, syscall.EPIPE) && !hasRetried {
+			log.Debugf("java attach hotspot pid %d/%d old hotspot JVM detected, require credential", h.pid, h.nsPid)
+			hasRetried = true
+			cleanConn()
+			cleanConn, err := h.connect(true)
+			if err != nil {
+				return err
+			}
+			defer cleanConn()
+			goto retry
```

Comment:
why using go to and not a function?

Reply:
I moved the logic in command as it would failed on the 1st write, but still relay on goto as it's more simple to read and old conn need to be closed beforehand

Reply:
moving the logic on the 1st write would failed as read can fail too later on some java (oracle) version